### ISSUE TITLE
[ENH] Min/Max for histograms computation

### ISSUE THREAD

**Issue by tguillemot** [scikit-image/scikit-image#2189]
## Description
I was wondering if there is war to better define `exposure.histogram`.

I usually compute the histogram between 0. and 1. (for float images) but in scikit-image the histogram is computed between the min and max. I'm pretty sure some user apply the histogram without seeing that problem creating possibly bugs (the docstring is not really explicit on that subject).

I propose to add two optional argument to histogram `min` and `max` giving the possibility to the user to compute between the min/max value he wants. If they are put to None, the histogram is computed in the same way as we have now.

If you're not agree with that, maybe it could be interesting to add a comment about that on the docstring ?

I'm in Scipy-2016, so I can propose do it during the spring if you want ?

**Comment by sciunto** [comment]
Just to say that I was also surprised by the current behaviour when I discovered the function.

**Comment by jni** [comment]
Whoa! I didn't know this either! I'd be inclined to consider this a bug, and default min/max to the dtype range. How about an "image_range" argument, which can take as values:

- 'image': determine the range from the input image (the current default)
- 'dtype': determine the range from the expected range of the images of that data type.
- (int, int): use the given range.

The new default would be 'dtype' in this case.

@tguillemot it would be great if you could make this contribution! @stefanv and I will probably be sprinting on the book, not scikit-image, but we can help you get this through!

**Comment by sciunto** [comment]
This function is used in thresholding.py and few examples. Normally changing the API should break the tests, but it's probably a good thing to check manually.

**Comment by tguillemot** [comment]
@sciunto Thanks I will check that. 

Today I need to help the guys of scikit-learn for the sprints.
If someone wants to do it today it's not a problem.
 If nobody wants, I will tomorrow.

**Comment by stefanv** [comment]
@jni I guess very much like this: http://scikit-image.org/docs/stable/api/skimage.exposure.html#rescale-intensity

**Comment by stefanv** [comment]
@tguillemot Sorry for not being very visible today, I had a few other things to take care of.  But I am around :)

**Comment by tguillemot** [comment]
@stefanv Don't worry, I have to work on scikit-learn today ;)

**Comment by stefanv** [comment]
I will be here tomorrow morning still.  See you around!
​

**Comment by jni** [comment]
@stefanv yes, `rescale_intensity` was my model API for that suggestion. =)
